steps/test/unit/tReorderMSData.cc

```diff
@@ -1,94 +1,115 @@
 #include <boost/test/unit_test.hpp>
+#include <fstream>
+#include <limits>
+
+#include "dp3/steps/Step.h"
 
 #include "tStepCommon.h"
-#include <dp3/steps/Step.h>
-#include "../../NullStep.h"
+
+#include "../../../common/ParameterSet.h"
+
 #include "../../InputStep.h"
+#include "../../NullStep.h"
 #include "../../ReorderMSData.h"
-#include <dp3/base/DPBuffer.h>
-#include <dp3/base/DPInfo.h>
-#include "../../../common/ParameterSet.h"
-#include "../../../common/StringTools.h"
 
-using dp3::base::DPBuffer;
-using dp3::base::DPInfo;
 using dp3::common::Fields;
 using dp3::common::ParameterSet;
 using dp3::steps::Step;
 
 BOOST_AUTO_TEST_SUITE(reorder)
 
-bool compareBinaries(const std::string& filename1,
-                     const std::string& filename2) {
-  std::ifstream file1(
-      filename1,
-      std::ifstream::ate | std::ifstream::binary);  // open file at the end
-  std::ifstream file2(
-      filename2,
-      std::ifstream::ate | std::ifstream::binary);  // open file at the end
+// Function to compare two binary files. If the files for comparison are Meta
+// file, the flag isMetaFile is set to true.
+void compareBinaryFiles(const std::string& reference_filename,
+                        const std::string& input_filename, bool isMetaFile) {
+  std::pair<std::istreambuf_iterator<char>, std::istreambuf_iterator<char>>
+      iterator_pair;
 
-  if (file1.fail() || file2.fail()) {
+  std::ifstream reference_file_ptr(reference_filename);
+  std::ifstream input_file_ptr(input_filename);
+
+  if (reference_file_ptr.fail() || input_file_ptr.fail()) {
     std::cout << "Failed to open file" << std::endl;
+    BOOST_CHECK(false);
   }
 
-  const std::ifstream::pos_type fileSize = file1.tellg();
-  if (fileSize != file2.tellg()) {
-    return false;  // different file size
+  // Meta files contain path information in the first line of the binary.
+  // Since the path representation differs (i.e absolute path vs relative path),
+  // we skip the path check and compare only the rest of the meta information.
+  if (isMetaFile) {
+    reference_file_ptr.ignore(std::numeric_limits<std::streamsize>::max(),
+                              '\n');
+    input_file_ptr.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+
+    if (reference_file_ptr.eof() || input_file_ptr.eof()) {
+      std::cout << "End of file reached" << std::endl;
+      BOOST_CHECK(false);
+    }
+  } else {
+    // If file sizes are unequal return false.
+    if (reference_file_ptr.tellg() != input_file_ptr.tellg()) {
+      std::cout << "Unequal binary file sizes" << std::endl;
+      BOOST_CHECK(false);
+    }
+
+    // Resetting file pointers to compare from the beginning.
+    reference_file_ptr.seekg(0);
+    input_file_ptr.seekg(0);
   }
 
-  file1.seekg(0);  // rewind
-  file2.seekg(0);  // rewind
-
-  std::istreambuf_iterator<char> begin1(file1);
-  std::istreambuf_iterator<char> begin2(file2);
+  // Setting the file stream to not to skip whitespaces.
```

Comment:
`// Setting the file stream to not to skip whitespaces.` -> `// Setting the file stream to not skip whitespaces.`

Reply:
Also are you 100% sure this is necessary? My previous comment was to check if its necessary or not and add this if it is (I'm not sure it is, but I haven't checked). 
Please double check and make sure. If it is necessary then it can stay as is, otherwise it should be removed.

Reply:
No need to use noskipws as the generated binaries do not have any whitespaces, as most of the binary is floating point numbers as a char stream. Even in the meta-file we skip the check on the file-paths as they are not expected to be the same.

Reply:
To be 100% clear here, there is no such thing as "whitespace" when reading in binary mode, "whitespace" by definition is only something that makes sense in character space, so yes binaries don't contain whitespace...

This does not however mean that using a stream that skips whitespace cannot be an issue, in a binary (filled with floats or otherwise) you may have a byte equal to `0` and if the code considers that a whitespace (because it thinks its dealing with a character stream and has no knowledge or concept of binary vs ascii) and has whitespace skipping turned on then it will skip that `0` even though that `0` is not strictly speaking whitespace...

What is or isn't considered whitespace is controlled by the locale so behaviour here can vary based on locate.

So in short or to be 100% clear, having whitespace skipped absolutely IS a problem and it absolutely IS a bug if its occurring. Even if we think that the data does not have "whitespace" in it.

---

Anyway to summarise, the issue here is not whether we want to skip whitespace or not, its about whether its necessary for us to pass `noskipws` in order to turn whitespace skipping off, or whether its already turned off by default.
I was unsure what the default was for the code you were using as I know it can be the other way around.

This is what I asked you to look up or check... (And then to add noskipws **_IF_** necessary)

To save time I've now just gone and looked it up myself and the default for other stream buffer iterators is to have `skipws` on but for `std::istreambuf_iterator` it is off by default; therefore we don't need to set this, the original code was fine. i.e. you can just remove the noskipws stuff, it wasn't necessary.